src/LayerEditor/ui/data/polygon_operation.py

```diff
@@ -265,12 +266,50 @@ def _get_lines_and_qtpoly(self, diagram, polygon):
             lines.append(line)
         qtpolygon.append(QtCore.QPointF(points[0].xy[0], -points[0].xy[1]))
         return lines, qtpolygon
-        
+
+    @staticmethod
+    def _get_wire_oriented_vertices(wire):
```

Comment:
Starting Pycharm to just improve a single comment I finally make also soma simplifications.
In particular np.append is quite inefficient as it needs to enlarge whole array for every append.
Timing for append of 1000 elements gives list.append 50x faster. Even worse for larger lists.